db/collection.go

```diff
@@ -661,10 +702,28 @@ func validateSourcesNotModified(
 			continue
 		}
 
-		// DeepEqual is temporary, as this validation is temporary, for example soon
-		// users will be able to be able to change the migration
-		if !reflect.DeepEqual(oldCol.Sources, newCol.Sources) {
-			return NewErrCollectionSourcesCannotBeMutated(newCol.ID)
+		newColSources := newCol.CollectionSources()
+		oldColSources := oldCol.CollectionSources()
+
+		if len(newColSources) != len(oldColSources) {
+			return NewErrCollectionSourcesCannotBeAddedRemoved(newCol.ID)
+		}
+
+		for i := range newColSources {
+			if newColSources[i].SourceCollectionID != oldColSources[i].SourceCollectionID {
+				return NewErrCollectionSourceIDMutated(
+					newCol.ID,
+					newColSources[i].SourceCollectionID,
+					oldColSources[i].SourceCollectionID,
+				)
+			}
+		}
+
+		newQuerySources := newCol.QuerySources()
+		oldQuerySources := oldCol.QuerySources()
+
+		if len(newQuerySources) != len(oldQuerySources) {
+			return NewErrCollectionSourcesCannotBeAddedRemoved(newCol.ID)
```

Comment:
todo: This validation is no longer doing what it was doing before. This means that the name is no longer accurate. Sources can be modified. We can't add or remove them or change their ID, be they can be modified. Maybe change the name to `validateSourcesModification`.

Reply:
Good spot, will do.

- [ ] Rename validation func